game-core/src/main/java/games/strategy/triplea/ai/pro/ProNonCombatMoveAi.java

```diff
@@ -878,6 +882,18 @@ private void moveUnitsToDefendTerritories(final List<ProTerritory> prioritizedTe
       // Determine if all defenses are successful
       boolean areSuccessful = true;
       boolean containsCapital = false;
+      final Set<Territory> territoriesToCheck = new HashSet<>();
+      for (final ProTerritory patd : territoriesToTryToDefend) {
```

Comment:
Noting that we have a matches operation, maybe that could benefit from parallelism? Not sure, FWIW a quick parallel version:
```
      territoriesToTryToDefend.parallelStream()
          .forEach(patd -> {
            final Territory t = patd.getTerritory();
            territoriesToCheck.add(t);
            final List<Unit> nonAirDefenders =
                CollectionUtils.getMatches(moveMap.get(t).getTempUnits(), Matches.unitIsNotAir());
            for (final Unit u : nonAirDefenders) {
              territoriesToCheck.add(unitTerritoryMap.get(u));
            }
          });
```

Reply:
Same as above.